tests/linearizability/lease_ids.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linearizability
+
+import (
+	"sync"
+)
+
+type clientId2LeaseIdMapper interface {
```

Comment:
What's the reason to map leases to clients? Would it be more interesting if we allowed every client to touch every lease? Changes needed:
* We just store available lease ids. Possibly with some limit to avoid explosion on to many LeaseGrant.
* When client needs to use lease it will take it out from storage to avoid racing with other clients. When client is done with a lease it can return it to storage, assuming it didn't revoke it.